### ISSUE TITLE
MIPSY stray comma error - breaks compiler

### ISSUE THREAD

**Issue by sourceofvegemite** [insou22/mipsy#124]
A stray comma after a line of code breaks the compiler with no error message.  Program hangs until closed with "ctrl + c".
![image](https://user-images.githubusercontent.com/95154624/143727102-f65838ba-a642-4e93-9cd6-c6a13a1d1df4.png)
![image](https://user-images.githubusercontent.com/95154624/143727093-de936a84-4f2b-4adf-91ac-8deceb38f60e.png)

Should there be some error message that pops up?  Thanks :D

**Comment by insou22** [comment]
I'm unfortunately not able to replicate this exact output:
```
$ cat foo.s
main:
	add	$t0, $t1, $t2,
$ 1521 mipsy
[mipsy] load foo.s
error: failed to compile `foo.s`
 --> ./foo.s:2:30
  |
2 |         add    $t0, $t1, $t2,
  |                               failed to parse

[mipsy]
```

regardless, this still is clearly not a desirable diagnostic, so I'll mark this as a bug.

If you're able to find steps to exactly reproduce that behaviour (where mipsy stops printing after `error: ...`), please update the issue!

**Comment by Dylan-Brotherston** [comment]
It's possible thay there is another issue causing the blocking.
@sourceofvegemite can you provide the entire file?

**Comment by sourceofvegemite** [comment]
[bugged_file.txt](https://github.com/insou22/mipsy/files/7614476/bugged_file.txt)
[working_file.txt](https://github.com/insou22/mipsy/files/7614477/working_file.txt)

Attached the files as txt.  I've tried to put commas everywhere else in the file, but I never end up with a warning error like you've shown.

I've also tried it with another file (attached below screenshot), and end up with the same problem.  I'm using VSCode SSH FS to connect to the CSE system.

![image](https://user-images.githubusercontent.com/95154624/143787778-b7e8261b-a95d-416c-a76d-97e434eb5cea.png)
[broken_pun.txt](https://github.com/insou22/mipsy/files/7614488/broken_pun.txt)

**Comment by insou22** [comment]
this is definitely a very weird one -- I'm now able to reproduce your behaviour (with the broken file you attached, thanks!) and I've reduced it to a potential MCVE:
```
$ cat -A mcve.s
x^I^I,$
$ 1521 mipsy mcve.s
error: failed to parse `mcve.s`
^C
```
note that those `^I`s are tab characters, and there needs to be at least 2 of them it seems, and nothing between the `,` and end-of-line (that $ signifies end of line).

will have to investigate further

**Comment by insou22** [comment]
Fixed in #128, thanks @joooooooooooooooooooooooooooooooooooosh !
